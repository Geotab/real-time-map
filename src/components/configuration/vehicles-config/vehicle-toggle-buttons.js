import React, { Component } from "react";
import { deviceSearch } from "./vehicle-search";

export class VehicleToggleButtons extends React.Component {
  constructor(props) {
    super(props);
    this.state = { visibility: true };
    this.toggleVisibility = this.toggleVisibility.bind(this);
  }

  toggleVisibility() {
    this.setState(state => {
      return { visibility: !state.visibility };
    });
  }

  render() {
    return (
      <>
        <button
          id="clearDevices"
          className={`toggleButton ${
            this.state.visibility ? "shown" : "notShown"
          }`}
          title="Toggle All Devices/Groups"
          onClick={() => {
            this.toggleVisibility();
            if (this.state.visibility) {
              deviceSearch.hideAll(this.props.setVehicleDisplay);
            } else {
              deviceSearch.showAll(this.props.setVehicleDisplay);
            }
          }}
        ></button>
        <button
          id="deleteDevices"
          className="deleteButton"
          title="Delete All Devices/Groups"
          onClick={() =>
            deviceSearch.deleteAllItems(this.props.setVehicleDisplay)
          }
        ></button>
      </>
    );
  }
}
