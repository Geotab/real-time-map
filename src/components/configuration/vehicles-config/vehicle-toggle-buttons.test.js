import React from "react";
import {VehicleToggleButtons} from "./vehicle-toggle-buttons";
import { render, mount, shallow } from "enzyme";

let wrapper;

let state;
beforeEach(() => {
  state = {
    visibility: true 
  };
});

test("should successfully render the component and match screenshot", () => {
  const component = render(<VehicleToggleButtons />);
  expect(component).toMatchSnapshot();
});

test("should successfully togglenpm te component state", () => {
  wrapper = shallow(<VehicleToggleButtons />);
  const instance = wrapper.instance();
  instance.toggleVisibility();
  expect(wrapper.state.visibility).toBeFalsy();
});