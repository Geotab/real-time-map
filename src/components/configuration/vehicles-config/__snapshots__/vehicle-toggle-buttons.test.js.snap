// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successfully render the component and match screenshot 1`] = `
Array [
  <button
    class="toggleButton shown"
    id="clearDevices"
    title="Toggle All Devices/Groups"
  />,
  <button
    class="deleteButton"
    id="deleteDevices"
    title="Delete All Devices/Groups"
  />,
]
`;
